linux/ibus-keyman/src/engine.c

```diff
@@ -804,16 +808,35 @@ ibus_keyman_engine_process_key_event(
     return FALSE;
   }
 
+  // #10476: Core currently doesn't handle IBUS_MOD{2-4}_MASK modifiers.
+  // On Ubuntu 23.10/24.04 we get IBUS_MOD4_MASK set on keycode 0x39 (space) when
+  // the user tries to switch keyboards. Since this is not a regular keypress
+  // and Core doesn't handle it, we need to just return and let the Gnome deal
+  // with it. We could consider to add it to Core and let Core ignore it.
+  // As for IBUS_MOD3_MASK it's unclear when/how that gets set, so we
+  // just not deal with that for now until we notice problems.
+  if (state & IBUS_MOD4_MASK) {  // Super/Meta/Windows key depressed
+    g_message("%s: Not handling keys with IBUS_MOD4_MASK (Super) modifier", __FUNCTION__);
+    return FALSE;
+  }
+
+  // REVIEW: Do we need to do something about IBUS_MOD2_MASK (NumLock)? I guess
+  // if a keyboard would have rules for the numeric keypad we might need
+  // it. At the moment however we let Core process the keypress and since
+  // it doesn't have rules for the numeric keypad keys we eventually
+  // forward the key to ibus (in process_emit_keystroke_action) and let
+  // Gnome deal with it.
```

Comment:
We do have a flag for numlock - but it is currently commented out. In theory Keyman .kmn keyboards support rules with NUMLOCK modifiers, but this is not recommended and has never really been tested.